homeassistant/components/rova/sensor.py

```diff
@@ -27,36 +31,47 @@
 SCAN_INTERVAL = timedelta(hours=12)
 
 
-class RovaSensorMetadata(NamedTuple):
+@dataclass
+class RovaSensorEntityDescription(SensorEntityDescription):
     """Metadata for an individual rova sensor."""
 
-    name: str
-    json_key: str
-    icon: str
+    json_key: str | None = None
 
+    def __post_init__(self) -> None:
+        """Check required keys are present."""
+        if self.json_key is None:
+            # pragma: no cover
+            raise TypeError
+        self._json_key = self.json_key
 
-SENSOR_TYPES: dict[str, RovaSensorMetadata] = {
-    "bio": RovaSensorMetadata(
-        "Biowaste",
-        json_key="gft",
+
+SENSOR_TYPES: tuple[RovaSensorEntityDescription, ...] = (
+    RovaSensorEntityDescription(
+        key="bio",
+        name="Biowaste",
         icon="mdi:recycle",
+        json_key="gft",
     ),
-    "paper": RovaSensorMetadata(
-        "Paper",
-        json_key="papier",
+    RovaSensorEntityDescription(
+        key="paper",
+        name="Paper",
         icon="mdi:recycle",
+        json_key="papier",
     ),
-    "plastic": RovaSensorMetadata(
-        "PET",
-        json_key="pmd",
+    RovaSensorEntityDescription(
+        key="plastic",
+        name="PET",
         icon="mdi:recycle",
+        json_key="pmd",
     ),
-    "residual": RovaSensorMetadata(
-        "Residual",
-        json_key="restafval",
+    RovaSensorEntityDescription(
+        key="residual",
+        name="Residual",
         icon="mdi:recycle",
+        json_key="restafval",
     ),
-}
+)
+MONITORED_CONDITIONS: list[str] = [desc.key for desc in SENSOR_TYPES]
```

Comment:
Had to add this list comprehension, as the dict keys where previously used for the `vol.In` call below.